packages/plugin-react/src/index.ts

```diff
@@ -112,20 +135,23 @@ export default function viteReact(opts: Options = {}): PluginOption[] {
         const isNodeModules = id.includes('/node_modules/')
         const isProjectFile =
           !isNodeModules && (id[0] === '\0' || id.startsWith(projectRoot + '/'))
+            ? !shouldProjectExclude(id)
+            : shouldProjectInclude(id)
```

Comment:
Would any "exclude" settings be ignored in this case?

Reply:
Good observation! With how this is currently written, one cannot `exclude` a subset of paths matched by `include`, which is certainly a valid use case.